### PR TITLE
Add emailverification

### PR DESCRIPTION
**Pull request by Kang-Quan**
Add email verification (only for users who use email + pw to sign up). Allow resending of email if they did not receive. Back button will bring user back to sign in page. Updated protected route to prevent user without verification to enter.

### CODE REVIEW

File: front-end/peer-prep/src/components/Auth/VerificationPage/styles.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
+export const VerificationPageContainerStyle = {
```
Comment (zenithyap) [inline front-end/peer-prep/src/components/Auth/VerificationPage/styles.tsx:1]: Should start with small letter ,verificationPageContainerStyle. Similarly for line 23.

File: front-end/peer-prep/src/components/Auth/VerificationPage/index.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+					>
+						resend
+					</Button>
+				</div>
+			</div>
+		</div>
+	);
+};
+
+const verifytext = "Thanks for joining peerprep";
```
Comment (zenithyap) [inline front-end/peer-prep/src/components/Auth/VerificationPage/index.tsx]: Just a suggestion, maybe you could capitalise peerprep to PeerPrep instead?

File: front-end/peer-prep/src/components/Auth/VerificationPage/index.tsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+const verifytextInner =
+	"To complete your profile we will need you to" +
```
Comment (zenithyap) [inline front-end/peer-prep/src/components/Auth/VerificationPage/index.tsx]: Add space after the word "to". Else it looks like this:

Screenshot 2023-10-17 at 1 21 43 PM

File: front-end/peer-prep/src/components/Auth/VerificationPage/styles.tsx (hunk 1 above)
Comment (zenithyap) [inline front-end/peer-prep/src/components/Auth/VerificationPage/styles.tsx]: Should start with small letter, verificationPageContainerStyle. Similarly for line 23.

**Review by zenithyap — CHANGES_REQUESTED** [review]
Other than a couple of small changes, looks good!

File: front-end/peer-prep/src/components/Auth/VerificationPage/index.tsx (hunk 2 above)
Comment (Kang-Quan) [reply front-end/peer-prep/src/components/Auth/VerificationPage/index.tsx]: thanks for pointing out, made the necessary changes

File: front-end/peer-prep/src/components/Auth/VerificationPage/index.tsx (hunk 3 above)
Comment (Kang-Quan) [reply front-end/peer-prep/src/components/Auth/VerificationPage/index.tsx]: thanks for pointing out, made the necessary changes

File: front-end/peer-prep/src/components/Auth/VerificationPage/styles.tsx (hunk 1 above)
Comment (Kang-Quan) [reply front-end/peer-prep/src/components/Auth/VerificationPage/styles.tsx]: thanks for pointing out, made the necessary changes

**Review by Acerizm — APPROVED** [review]
LGTM! Thank you for coming up with an awesome solution that only allows users that sign up with their email to reconfirm that they are the ones signing up for our app!

**Comment by zenithyap** [comment]
Seems to work fine now, thanks for the changes!

**Review by zenithyap — APPROVED** [review]
Seems to work now, thanks for the changes!

**Pull request merged**